projects/scion/toolkit/operators/src/operators.spec.ts

```diff
@@ -12,14 +12,82 @@ import {BehaviorSubject, Observable, Observer, of, Subject, TeardownLogic} from
 import {fakeAsync, flushMicrotasks, TestBed} from '@angular/core/testing';
 import {NgZone} from '@angular/core';
 import {finalize, tap} from 'rxjs/operators';
-import {combineArray, mapArray, observeInside, subscribeInside} from './operators';
+import {combineArray, filterArray, mapArray, observeInside, subscribeInside} from './operators';
 import {ObserveCaptor} from '@scion/toolkit/testing';
 import {bufferUntil, distinctArray} from '@scion/toolkit/operators';
 
 describe('Operators', () => {
 
   beforeEach(() => TestBed.configureTestingModule({}));
 
+  describe('filterArray', async () => {
+
+    it('should filter items', async () => {
+      const observeCaptor = new ObserveCaptor();
+
+      of(['a', 'b', 'c'])
+        .pipe(filterArray(item => item === 'a'))
+        .subscribe(observeCaptor);
+
+      await observeCaptor.waitUntilCompletedOrErrored();
+      await expect(observeCaptor.getLastValue()).toEqual(['a']);
+    });
+
+    it('should not filter items if predicate is undefined', async () => {
+      const observeCaptor = new ObserveCaptor();
+
+      of(['a', 'b', 'c'])
+        .pipe(filterArray(undefined))
+        .subscribe(observeCaptor);
+
+      await observeCaptor.waitUntilCompletedOrErrored();
+      await expect(observeCaptor.getLastValue()).toEqual(['a', 'b', 'c']);
+    });
+
+    it('should not filter `null` values', async () => {
+      const observeCaptor = new ObserveCaptor();
+
+      of(['a', null, null])
+        .pipe(filterArray(undefined))
+        .subscribe(observeCaptor);
+
+      await observeCaptor.waitUntilCompletedOrErrored();
+      await expect(observeCaptor.getLastValue()).toEqual(['a', null, null]);
+    });
+
+    it('should not filter `undefined` values', async () => {
+      const observeCaptor = new ObserveCaptor();
+
+      of(['a', undefined, undefined])
+        .pipe(filterArray(undefined))
```

Comment:
Please pass an "always-true" predicate instead of `undefined` and change the testee to `['a', undefined, undefined, 'b']`.